### ISSUE TITLE
Upcase method name override 

### ISSUE THREAD

**Issue by pcardoso** [flightphp/core#173]
Hello, 

When using _method or HTTP_X_HTTP_METHOD_OVERRIDE, should the getMethod return value in uppercase?  The browser does this for GET/POST. 

In the current version the following example doesn't match, unless I specify both methods with the same case.

``` html
   <form action="/users/123" method="post" role="form">
    <input type="hidden" name="_method" value="put">
```

``` php
Flight::route('PUT /users/@id', function($id) { ...
```

Also, shouldn't method override only apply when the main method is POST?

I can fix this and submit a PR, I just wanted to check things first. Thanks.

**Comment by mikecao** [comment]
You are right it should be uppercase. Fixed in commit a32f2c2087258c8f3f3e58bc95d73efa1c8043c6.

Ideally you should only override the method in a POST but I didn't want to limit people's options. If you want to override GET to something else, you can do it.

**Comment by pcardoso** [comment]
Fair enough! Thanks!
